jlm/rvsdg/simple-node.hpp

```diff
@@ -120,6 +134,31 @@ SimpleNode::output(size_t index) const noexcept
   return static_cast<simple_output *>(Node::output(index));
 }
 
+struct CreateOpNodeResult
+{
+  Node * node;
+  std::vector<output *> outputs;
+};
+
+template<typename OpType, typename... OpParameters>
+CreateOpNodeResult
+CreateOpNode(const std::vector<output *> & arguments, OpParameters... opParameters)
+{
+  Region * region = arguments.at(0)->region();
+  for (const auto & argument : arguments)
+  {
+    JLM_ASSERT(argument->region() == region);
+  }
```

Comment:
This can be removed as it is checked in the constructor of the input: https://github.com/phate/jlm/blob/6db940a6220bd6be1ba2203edceb6269114ea4b6/jlm/rvsdg/node.cpp#L24C1-L40C2

Reply:
removed redundant checking